### DISCUSSION TITLE
support iOS17 for Linux

### DISCUSSION THREAD

**Discussion by Rodionp13** [doronz88/pymobiledevice3#542] [Q&A] [answered]
Good day. Dear @doronz88 , I've started to study your documentation and source code recently. Sorry, if I'm wrong in some assumptions. (For this magic is quite hard for me to understand yet.=))
Are you going to support iOS17 for Linux ? And what are the problems (if there are any) you've faced currently with?
Does the absence of `remoted` and `remotepairingd` on Linux is the key problem or there's smth else? (Sorry if I'm completely wrong) 
Thanks in advance !

**Comment by doronz88** [comment, marked as answer]
pymobiledevice3 currently re-implements the entirety of `remoted` and `remotepairingd`. The problem on linux exists with USB Ethernet driver. Apple implemented the controller in some non-standard way, making it not work on other platforms. Some cool kids on the [Discord](https://discord.gg/52mZGC3JXJ) figured out some stuff about it (there's on ongoing thread) but I don't know when a solution will be found.

**Reply by Rodionp13** [reply]
Got it, thanks!

**Comment by Rodionp13** [comment]
One more question has in mind(it's not related with main theme however). Could you please be a bit more specific about this function `stop_remoted` ? It's the part of `get_device_list`. Why do you need to suspend & resume the daemon ?

**Reply by doronz88** [reply]
Because the `remoted` on the device-side cannot handle two connections from same IP at once - so it closes the `remoted`'s FD, leading it to re-establish the connection and then close our own FD - meaning it fights us over the connection.

**Reply by Rodionp13** [reply]
thanks!
